lib/src/ui/pages/settings/settings_notifier.dart

```diff
@@ -0,0 +1,72 @@
+import 'package:allthenews/src/domain/appinfo/app_info_repository.dart';
+import 'package:allthenews/src/domain/common/function.dart';
+import 'package:allthenews/src/domain/settings/popular_news_criterion.dart';
+import 'package:allthenews/src/domain/settings/settings_repository.dart';
+import 'package:allthenews/src/domain/settings/app_theme.dart';
+import 'package:allthenews/src/ui/common/theme/theme_notifier.dart';
+import 'package:flutter/material.dart';
+
+class SettingsNotifier extends ChangeNotifier {
+  final AppInfoRepository _appInfoRepository;
+  final SettingsRepository _settingsRepository;
+
+  SettingsNotifier(this._settingsRepository, this._appInfoRepository);
+
+  SettingsViewState _viewState = SettingsViewState(
+    isLoading: true,
+    appVersion: null,
+    isDarkModeEnabled: false,
+    selectedPopularNewsCriterion: PopularNewsCriterion.viewed,
+  );
+
+  SettingsViewState get viewState => _viewState;
+
+  void loadSettings() async {
+    final appTheme = _settingsRepository.getTheme();
+    final appVersion = _appInfoRepository.getAppVersion();
+    final selectedCriterion = _settingsRepository.getPopularNewsCriterion();
+    _viewState = SettingsViewState(
+      isLoading: false,
+      appVersion: await appVersion,
+      isDarkModeEnabled: await appTheme == AppTheme.dark,
+      selectedPopularNewsCriterion: await selectedCriterion,
+    );
+    notifyListeners();
+  }
+
+  void selectDarkMode(bool isSelected, Read read) async {
```

Comment:
przekazywanie samej funkcji po to, żeby uniknąć przekazywania kontekstu, dobrze rozumiem?

Reply:
tak, nie chciałem przesyłać całego kontekstu tylko powiadomić nadrzędnego ThemeNotifier z poziomu SettingsNotifier, tak, żeby ta logika nie była na UI, więc zawęziłem to do przesłania referencji do context.read :)